applications/nrf_desktop/src/modules/hids.c

```diff
@@ -487,20 +496,50 @@ static void notify_secured_fn(struct k_work *work)
 	}
 }
 
+static void broadcast_hids_subscriber_state(void *subscriber, bool enabled)
+{
+	struct hid_report_subscriber_event *event = new_hid_report_subscriber_event();
+
+	event->subscriber = subscriber;
+
+	if (enabled) {
+		event->params.pipeline_size = HIDS_SUBSCRIBER_PIPELINE_SIZE;
+		event->params.priority = HIDS_SUBSCRIBER_PRIORITY;
+		event->params.report_max = HIDS_SUBSCRIBER_REPORT_MAX;
+	}
+
+	event->connected = enabled;
+
+	APP_EVENT_SUBMIT(event);
+}
+
 static void notify_hids(const struct ble_peer_event *event)
 {
 	int err = 0;
+	static bool subscriber_connected;
 
 	switch (event->state) {
 	case PEER_STATE_CONNECTED:
 		__ASSERT_NO_MSG(cur_conn == NULL);
 		cur_conn = event->id;
 		err = bt_hids_connected(&hids_obj, event->id);
-
 		if (err) {
 			LOG_ERR("Failed to notify the HID Service about the"
 				" connection");
 		}
+
+		__ASSERT_NO_MSG(!subscriber_connected);
+
+		broadcast_hids_subscriber_state(event->id, true);
+
+		subscriber_connected = true;
+		break;
+
+	case PEER_STATE_DISCONNECTING:
+		__ASSERT_NO_MSG(subscriber_connected);
```

Comment:
Can we perform actions only if `subscriber_connected` is set instead of asserting to avoid asserting on multiple `PEER_STATE_DISCONNECTING`?

Reply:
We can. Done